db/migrate/20240909183413_create_categories.rb

```diff
@@ -0,0 +1,44 @@
+class CreateCategories < ActiveRecord::Migration[7.1]
+  def change
+    create_table :detectors do |t|
+      t.string :name
+      t.float :confidence
+
+      t.timestamps
+    end
+    add_index :detectors, :name, unique: true
+
+    create_table :categories do |t|
+      t.string :name
+      t.text :description
+
+      t.timestamps
+    end
+    add_index :categories, :name, unique: true
+
+    create_table :detector_categories do |t|
```

Comment:
we likely want an index that is on the combined ids of the two tables this joins to prevent tables scans. Possibly in both directions depending on our future expected needs (do we need all categories for a specific detector only, or are the times when we want all detectors for a specific category?)

Reply:
Is there a cost to building both right now, just in case? I'm not clear right now what we'd need. During categorization, starting from a given term, it would probably be "take the detectors for which this term activated, and increment their associated category scores" - which feels like a bit like "find the categories associated with this detector". For the final population of category scores, though direction might be the other way - I'm not sure yet.

Reply:
Minimal cost for having both. (In fact the two tables will both be so small that even not having these indexes would be fine... but I didn't say that!)